packages/components/src/components/SquiggleOutputViewer/index.tsx

```diff
@@ -0,0 +1,141 @@
+  isRunning: boolean;
+  editor?: CodeEditorHandle;
+} & PartialPlaygroundSettings;
+
+/* Wrapper for SquiggleViewer that shows the rendering stats and isRunning state. */
+export const SquiggleOutputViewer = forwardRef<SquiggleViewerHandle, Props>(
+  ({ squiggleOutput, isRunning, editor, ...settings }, viewerRef) => {
+    const resultItem = getResultValue(squiggleOutput);
+    const resultVariables = getResultVariables(squiggleOutput);
+
+    const hasResult = Boolean(resultItem?.ok);
+    const variablesCount = resultVariables?.ok
+      ? resultVariables.value.value.entries().length
+      : 0;
+
+    const [mode, setMode] = useState<"variables" | "result">(
+      hasResult ? "result" : "variables"
+    );
+
+    let squiggleViewer: JSX.Element | null = null;
+    if (squiggleOutput.code) {
+      let usedResult: result<SqValue, SqError> | undefined;
+      switch (mode) {
+        case "result":
+          usedResult = resultItem;
+          break;
+        case "variables":
+          usedResult = resultVariables;
+      }
+
+      if (usedResult) {
+        squiggleViewer = usedResult.ok ? (
+          <div className="relative">
+            {isRunning && (
+              // `opacity-0 squiggle-semi-appear` would be better, but won't work reliably until we move Squiggle evaluation to Web Workers
+              <div className="absolute z-10 inset-0 bg-white opacity-50" />
+            )}
+            <ErrorBoundary>
+              <SquiggleViewer
+                {...settings}
+                ref={viewerRef}
+                value={usedResult.value}
+                editor={editor}
+              />
+            </ErrorBoundary>
+          </div>
+        ) : (
+          <SquiggleErrorAlert error={usedResult.value} />
+        );
+      }
+    }
+
+    return (
+      <Layout
+        menu={
+          <Dropdown
+            render={({ close }) => (
+              <DropdownMenu>
+                <DropdownMenuActionItem
+                  icon={CodeBracketIcon}
+                  title={
+                    <MenuItemTitle
+                      title="Variables"
+                      type={variablesCount ? `{}${variablesCount}` : null}
+                    />
+                  }
+                  onClick={() => {
+                    setMode("variables");
+                    close();
+                  }}
+                />
+                <DropdownMenuActionItem
+                  icon={CodeBracketIcon}
+                  title={
+                    <MenuItemTitle
+                      title="Result"
+                      type={hasResult ? "" : null}
+                    />
+                  }
+                  onClick={() => {
+                    setMode("result");
+                    close();
+                  }}
+                />
+              </DropdownMenu>
+            )}
+          >
+            <Button size="small">
+              {mode === "variables" ? "Variables" : "Result"}
```

Comment:
Add down-triangle too.

Reply:
No problem, but note that it won't look identically to `Save` (no vertical bar). Save button acts differently based on where you click.